rsc/client/opendota/match.go

```diff
@@ -0,0 +1,58 @@
+package opendota
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+var (
+	ErrFetchMatchDetail = fmt.Errorf("failed to fetch match detail")
+)
+
+type OpenDotaAPI struct {
+	httpClient *http.Client
+	APIURL     string
```

Comment:
maybe `BaseURL` or `Host` is a better name, but what do I know, naming is hard.

Reply:
Ahh correct, its look better than `APIURL`. I'll pick BaseURL 👍